16_wordle/main.go

```diff
@@ -1,21 +1,71 @@
 package main
 
 import (
+	"bytes"
 	"fmt"
+	"log"
+	"os"
 )
 
+// func valid(input []byte) bool {
+// 	if bytes.Contains(input []byte) {
+// 		return true
+// 	}
+// 	return false
+// }
+
 func main() {
-	var input string
+	var input []byte
+
+	data, err := os.ReadFile("word_list/word_list")
+	if err != nil {
+		log.Fatal(err)
+	}
+	// os.Stdout.Write(data)
+	words := bytes.Split([]byte(data), []byte("\n"))
+	// debug
+	// fmt.Printf("%q\n", formattedData)
+	// fmt.Printf("%q\n", bytes.Split([]byte(data), []byte("\n")))
+	// fmt.Println(bytes.Contains([]byte("brown"), formattedData))
+	chosenWord := "brown"
+	chosenWordArray := []byte(chosenWord)
 
 	for i := 0; ; i++ {
-		fmt.Printf("Enter 5 digit word\n>")
+		fmt.Printf("Enter 5 letter word\n>")
 		fmt.Scan(&input)
-
-		if input == "Test" {
-			fmt.Print("Correct!\n")
+		// check if word on list
+		if isValid(input, words) && isChosen(input, chosenWordArray) {
 			break
-		} else {
-			fmt.Print("Not correct, try again!\n")
 		}
+		if isValid(input, words) && !isChosen(input, chosenWordArray) {
+			fmt.Print("But not chosen word, try again!\n")
+		}
```

Comment:
See if you can simplify this dual condition.

E.g. https://go.dev/play/p/iwJ4yG4Z8lZ